autogen/tools/experimental/perplexity/perplexity_search.py

```diff
@@ -0,0 +1,67 @@
+# Copyright (c) 2023 - 2025, AG2ai, Inc., AG2ai open-source projects maintainers and core contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Annotated, Any, Dict, List, Optional
+
+import requests
+
+from autogen.tools import Tool
+
+from .schemas import PerplexityChatCompletionResponse, SearchResponse
+
+
+class PerplexitySearchTool(Tool):
+    def __init__(
+        self,
+        model: str = "sonar",
+        api_key: Optional[str] = None,
+        max_tokens: int = 1000,
+        search_domain_filter: Optional[List[str]] = None,
+    ):
```

Comment:
Can you please add docstrings to your functions so they can be used in documentation and for future contributors.

```suggestion
    ):
        """
        Perplexity AI search tool for web search.

        Args:
            model (str): The model to use for the search. Default is "sonar". See https://docs.perplexity.ai/guides/model-cards for model options.
            api_key (str): The Perplexity API key. If not provided, it will be read from the environment variable PERPLEXITY_API_KEY.
            max_tokens (int): The maximum number of tokens to generate in the response. Default is 1000.
            search_domain_filter (list[str]): A list of search domains to filter the search results. Default is None. E.g. ["arxiv.org", "nature.com"]
        """
```